CMakeLists.txt

```diff
@@ -232,6 +236,92 @@ endif()
 
 include(cmake/optional.cmake)
 
+set(core_cxx_flags)
+# Don't allow extended (non-ASCII) symbols in identifiers. This is easier for code review.
+try_append_cxx_flags(core_cxx_flags "-fno-extended-identifiers")
+
+# Currently all versions of gcc are subject to a class of bugs, see the
+# gccbug_90348 test case (only reproduces on GCC 11 and earlier) and
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111843. To work around that, set
+# -fstack-reuse=none for all gcc builds. (Only gcc understands this flag).
+try_append_cxx_flags(core_cxx_flags "-fstack-reuse=none")
+
+separate_arguments(core_cxx_flags)
+target_compile_options(core INTERFACE ${core_cxx_flags})
+unset(core_cxx_flags)
+
+if(HARDENING)
+  if(MSVC)
+    try_append_linker_flag(core "/DYNAMICBASE")
+    try_append_linker_flag(core "/HIGHENTROPYVA")
+    try_append_linker_flag(core "/NXCOMPAT")
+  else()
+    set(fortfy_flags "-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3")
```

Comment:
I wonder if these could work the same way as what we discussed with hardening? Add debug/release interface libs?

Reply:
Reworked.